.ycm_extra_conf.py

```diff
@@ -36,7 +36,7 @@
 # CHANGE THIS LIST OF FLAGS. YES, THIS IS THE DROID YOU HAVE BEEN LOOKING FOR.
 flags = [
     '-DNDEBUG',
-    '-DWITH_TBB',
+    '-DNMC_HAVE_TBB',
```

Comment:
I see your reasoning. However, currently, these two naming schemes are also used in different "namespaces". You are using `NMC_HAVE_X` for C++ preprocessor variables, and `NMC_WITH_X` for CMake variables. This is because we have a policy of having a hard error if the user requests a feature, say CUDA or OpenMP, instead of "silently" building with the feature disabled. This saves us from lots of silent "MPI isn't working when I compile the MPI target" errors.